engine/collection/ingest/engine.go

```diff
@@ -224,14 +225,20 @@ func (e *Engine) onTransaction(originID flow.Identifier, tx *flow.TransactionBod
 	// or a different cluster)
 	if originID == e.me.NodeID() {
 
-		log.Debug().Msg("propagating transaction to cluster")
+		// if the target cluster has no collection node other than possibly this
+		// collection node then don't attempt to propagate the transaction further
+		recipientIDs := txCluster.Filter(filter.Not(filter.HasNodeID(e.me.NodeID())))
+		if len(recipientIDs) > 0 {
```

Comment:
Can we put this check in the network layer?

If the `Multicast` is asked to send to an empty list of node IDs, it wouldn't make any networking request.

Reply:
so the network layer has the check actually - https://github.com/onflow/flow-go/blob/master/network/p2p/network.go#L406
It probably needs to send a specific error type when the number of target ids are 0 (after the application of the filter).
Ok, lemme make that change - that would be a better solution.

Reply:
I think either the network layer should _not_ return an error when given an empty list (it would be an allowed input resulting in a no-op) or we should just keep this as-is.

Having the caller check for an empty list is equivalent to checking for a particular error return -- I don't see the need for that change.

```
if len(list) > 0 {
    multicast()
}
```

vs

```
err := multicast()
if network.IsEMptyListError(err) {
   // pass
}
````

Reply:
If we do the second option that you suggested, then the collection engine does not has to do the list filtering (which will be additionally done by the networking layer to remove `self` to avoid sending messages to itself).

Reply:
👍 Right, good point. That's better!